### PR TITLE
Closes #73

### LINKED ISSUE #73: Blog Post: Date Functions
Release Post Date: August 21

Covers the date/time functions

### PR DESCRIPTION
**Pull request by manciniedoardo**
Closes #73

### CODE REVIEW

File: posts/2023-08-21_date_functions_and_imputation/date_functions_and_imputation.qmd [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+`{admiral}` provides the following functions for date/datetime imputation:
+
+- `derive_vars_dt()`: Adds a date variable and a date imputation flag variable (optional) based on a --DTC variable and imputation rules.
+- `derive_vars_dtm()`: Adds a datetime variable, a date imputation flag variable, and a time imputation flag variable (both optional) based on a --DTC variable and imputation rules.
+- `impute_dtc_dtm()`: Returns a complete ISO 8601 datetime or `NA` based on a partial ISO 8601 datetime and imputation rules.
+- `impute_dtc_dt()`: Returns a complete ISO 8601 date (without time) or `NA` based on a partial ISO 8601 date(time) and imputation rules.
+- `convert_dtc_to_dt()`: Returns a date if the input ISO 8601 date is complete. Otherwise, `NA` is returned.
+- `convert_dtc_to_dtm()`: Returns a datetime if the input ISO 8601 date is complete (with missing time replaced by `"00:00:00"` as default). Otherwise, NA is returned.
+- `compute_dtf()`: Returns the date imputation flag.
+- `compute_tmf()`: Returns the time imputation flag.
```
Comment (bms63) [inline posts/2023-08-21_date_functions_and_imputation/date_functions_and_imputation.qmd:45]: Could we link all these functions to their admiral home page?

File: posts/2023-08-21_date_functions_and_imputation/date_functions_and_imputation.qmd [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+For a detailed look at the Imputation rules applied by these `{admiral}` functions, please visit [this vignette](https://pharmaverse.github.io/admiral/cran-release/articles/imputation.html#imputation-rules) on the documentation website.
+
+# Action Examples
+
+## Creating an Imputed Datetime and Date Variable and Imputation Flag Variables
+
+As described previously, `derive_vars_dtm()` derives an imputed datetime variable and the corresponding date and time imputation flags. The imputed date variable can then be derived by using `derive_vars_dtm_to_dt()`. It is not necessary and advisable to perform the imputation for the date variable if it was already done for the datetime variable. CDISC considers the datetime and the date variable as two representations of the same date. Thus the imputation must be the same and the imputation flags are valid for both the datetime and the date variable.
+
+```{r}
```
Comment (bms63) [inline posts/2023-08-21_date_functions_and_imputation/date_functions_and_imputation.qmd:57]: We should suppress startup messages and warnings 
image

File: posts/2023-08-21_date_functions_and_imputation/date_functions_and_imputation.qmd [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+# renv::use(lockfile = "renv.lock")
+
+library(admiraldev)
+```
+
+<!--------------- post begins here ----------------->
+
+# Introduction
+
+Date and time is collected in SDTM as character values using the extended [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format `"yyyy-dd-mmThh:mm:ss"`. This universal format allows missing parts date or time - e.g. the string`"2019-10"` represents a date where the day and the time are unknown. In contrast, ADaM timing variables like `ADTM` (Analysis Datetime) or `ADY` (Analysis Relative Day) are numeric variables, which can be derived only if the date or datetime is complete.
```
Comment (bms63) [inline posts/2023-08-21_date_functions_and_imputation/date_functions_and_imputation.qmd:28]:
```suggestion
Date and time is collected in SDTM as character values using the extended [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format `yyyy-dd-mmThh:mm:ss`. This universal format allows missing parts date or time - e.g. the string`"2019-10"` represents a date where the day and the time are unknown. In contrast, ADaM timing variables like `ADTM` (Analysis Datetime) or `ADY` (Analysis Relative Day) are numeric variables, which can be derived only if the date or datetime is complete. 
```

File: posts/2023-08-21_date_functions_and_imputation/date_functions_and_imputation.qmd [hunk 4]
```diff
@@ -0,0 +1,252 @@
+---
+title: "Date/Time Functions and Imputation in {admiral} "
+author:
+  - name: Edoardo Mancini
+description: "Dates, times and imputation can be a frustrating facet of any programming language. Here's how {admiral} makes all of this easy!"
+date: "2023-08-21"
+# please do not use any non-default categories.
+# You can find the default categories in the repository README.md
+categories: [admiral]
```
Comment (bms63) [inline posts/2023-08-21_date_functions_and_imputation/date_functions_and_imputation.qmd:9]: Lets add ADaM and mabye create a new tag for Date/Time?  We can the Date/Time to the other blog posts.

File: posts/2023-08-21_date_functions_and_imputation/date_functions_and_imputation.qmd [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+- `compute_tmf()`: Returns the time imputation flag.
+
+From the point of view of a typical ADaM programmer, the functions `impute_*`, `convert_*` and `compute_*` above can be viewed as utilities for treating dates and/or imputation within any custom code. In contrast, their `derive_*` find their use in directly deriving new timing variables and/or carrying out imputation at an ADaM dataset scale.
+
+For a detailed look at the Imputation rules applied by these `{admiral}` functions, please visit [this vignette](https://pharmaverse.github.io/admiral/cran-release/articles/imputation.html#imputation-rules) on the documentation website.
+
+# Action Examples
+
+## Creating an Imputed Datetime and Date Variable and Imputation Flag Variables
+
```
Comment (bms63) [inline posts/2023-08-21_date_functions_and_imputation/date_functions_and_imputation.qmd:54]: What about starting off with a really easy example that uses the `impute_` functions and discuss the arguments.  Then segue into how the `derive_` are wrappers around the `impute` and `compute` functions?

File: posts/2023-08-21_date_functions_and_imputation/date_functions_and_imputation.qmd [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+  "2019-04-12",
+  "2010-09",
+  NA_character_
+) %>%
+  derive_vars_dtm(
+    dtc = AESTDTC,
+    new_vars_prefix = "AST",
+    highest_imputation = "M",
+    date_imputation = "first",
+    time_imputation = "first"
```
Comment (bms63) [inline posts/2023-08-21_date_functions_and_imputation/date_functions_and_imputation.qmd:75]: I think a quick mention of these arguments and their options is warranted - if the `impute` function is not discussed.

**Review by bms63 — COMMENTED** [review]
Overall - really awesome post!!  Just a couple of thoughts on the content and presentation.  Maybe we can get this out by Tuesday?

**Review by kaz462 — APPROVED** [review]
Super useful summary! thanks @manciniedoardo for a great article!

File: posts/2023-08-21_date_functions_and_imputation/date_functions_and_imputation.qmd (hunk 6 above)
Comment (manciniedoardo) [reply posts/2023-08-21_date_functions_and_imputation/date_functions_and_imputation.qmd:150]: impute function discussed in new section, so i didn't action this.

**Pull request merged**